src/components/register-participant-modal/modal.tsx

```diff
@@ -0,0 +1,146 @@
+"use client"
+
+import { Button } from "../button"
+import { Input } from "../input"
+import { ModalContent } from "../modal"
+import {
+  User,
+  AtSign,
+  Github,
+  ArrowLeft,
+  CheckCircle,
+  XCircle,
+} from "lucide-react"
+
+import { useForm, SubmitHandler } from "react-hook-form"
+import { zodResolver } from "@hookform/resolvers/zod"
+import * as z from "zod"
+import { useEffect, useState } from "react"
+
+const formSchema = z.object({
```

Comment:
This schema should be in `./src/shared/types/participants.ts`, and the type `RegisterParticipant` should be infered from that and exported.